src/interfaces.ts

```diff
@@ -139,13 +140,56 @@ export interface ConfigFileInfo {
   dtsFiles?: string[];
 }
 
+interface CacheWithRedirects<T> {
+  ownMap: Map<string, T>;
+  redirectsMap: Map<typescript.Path, Map<string, T>>;
+  getOrCreateMapOfCacheRedirects(
+    redirectedReference: typescript.ResolvedProjectReference | undefined
+  ): Map<string, T>;
+  clear(): void;
+  setOwnOptions(newOptions: typescript.CompilerOptions): void;
+  setOwnMap(newOwnMap: Map<string, T>): void;
+}
+interface PerModuleNameCache {
+  get(
+    directory: string
+  ): typescript.ResolvedModuleWithFailedLookupLocations | undefined;
+  set(
+    directory: string,
+    result: typescript.ResolvedModuleWithFailedLookupLocations
+  ): void;
+}
+export interface ModuleResolutionCache
+  extends typescript.ModuleResolutionCache {
+  directoryToModuleNameMap: CacheWithRedirects<
+    Map<string, typescript.ResolvedModuleWithFailedLookupLocations>
+  >;
+  moduleNameToDirectoryMap: CacheWithRedirects<PerModuleNameCache>;
+  clear(): void;
+  update(compilerOptions: typescript.CompilerOptions): void;
+  getPackageJsonInfoCache?(): any;
+}
+// Till api is updated
```

Comment:
So this will be accessible in the TypeScript API in future?

Reply:
I think this may be it: https://github.com/microsoft/TypeScript/blob/74993a2a64bb2e423b40204bb54ff749cdd4ef54/src/compiler/moduleNameResolver.ts#L458

Reply:
```suggestion
// Until the API has been released and ts-loader is built against a version of TypeScript that contains it - see https://github.com/microsoft/TypeScript/blob/74993a2a64bb2e423b40204bb54ff749cdd4ef54/src/compiler/moduleNameResolver.ts#L458
```